R/simulate_infections.R

```diff
@@ -1,19 +1,191 @@
+##'   are missing, it will be assumed that R stays the same
+##' @param initial_infections numeric; the initial number of infections.
+##' @param day_of_week_effect either `NULL` (no day of the week effect) or a
+##'   numerical vector of length specified in [obs_opts()] as `week_length`
+##'   (default: 7) if `week_effect` is set to TRUE. Each element of the vector
+##'   gives the weight given to reporting on this day (normalised to 1).
+##'   The default is `NULL`.
+##' @param estimates deprecated; use [forecast_infections()] instead
+##' @param ... deprecated; only included for backward compatibility
+##' @inheritParams estimate_infections
+##' @inheritParams rt_opts
+##' @importFrom lifecycle deprecate_warn
+##' @importFrom data.table data.table merge.data.table nafill rbindlist
+##' @return A data.table of simulated infections (variable `infections`) and
+##'   reported cases (variable `reported_cases`) by date.
+##' @author Sebastian Funk
+##' @export
+#' @examples
+#' \donttest{
+#'   R <- data.frame(
+#'     date = seq.Date(as.Date("2023-01-01"), length.out = 14, by = "day"),
+#'     R = c(rep(1.2, 7), rep(0.8, 7))
+#'   )
+#'   sim <- simulate_infections(
+#'     R = R,
+#'     initial_infections = 100,
+#'     generation_time = generation_time_opts(
+#'       fix_dist(example_generation_time)
+#'     ),
+#'     delays = delay_opts(fix_dist(example_reporting_delay)),
+#'     obs = obs_opts(family = "negbin")
+#'   )
+#' }
+simulate_infections <- function(estimates, R, initial_infections,
+                                day_of_week_effect = NULL,
+                                generation_time = generation_time_opts(),
+                                delays = delay_opts(),
+                                truncation = trunc_opts(),
+                                obs = obs_opts(),
+                                CrIs = c(0.2, 0.5, 0.9),
+                                pop = 0, ...) {
+  ## deprecated usage
+  if (!missing(estimates)) {
     deprecate_warn(
       "2.0.0",
-      "simulate_infections()",
+      "simulate_infections(estimates)",
       "forecast_infections()",
-      "A new [simulate_infections()] function for simulating from given ",
-      "parameters is planned for implementation in the future."
+      details = paste0(
+        "This `estimates` option will be removed from [simulate_infections()] ",
+        "in version 2.1.0."
+      )
     )
-   forecast_infections(...)
+    return(forecast_infections(estimates = estimates, ...))
+  }
+
+  ## check inputs
+  assert_data_frame(R)
+  assert_names(
+    colnames(R),
+    must.include = c("date", "R")
+  )
+  assert_date(R$date, any.missing = FALSE)
+  assert_numeric(R$R, lower = 0)
+  assert_numeric(initial_infections, lower = 0)
+  assert_numeric(day_of_week_effect, lower = 0, null.ok = TRUE)
+  assert_numeric(pop, lower = 0)
+  assert_class(delays, "delay_opts")
+  assert_class(obs, "obs_opts")
+  assert_class(generation_time, "generation_time_opts")
```

Comment:
We should probably bundle these checks into one as it's getting repeated in several places.

Reply:
yes agree though that feels like its own issue?

Reply:
It is now: #559